apps/bfd-model/bfd-model-rif/src/main/resources/db/scripts/BFD-766-fix-beneficiary-monthly-data/05_process_ccw_table.sql

```diff
@@ -90,256 +90,324 @@ DECLARE
-                rcd.BENE_ID,
-                TO_DATE(rcd.RFRNC_YR || '-11-01', 'YYYY-MM-DD'),
-                rcd.FIPS_STATE_CNTY_NOV_CD,
-                rcd.MDCR_STUS_NOV_CD,
-                rcd.MDCR_ENTLMT_BUYIN_11_IND,
-                rcd.HMO_11_IND,
-                rcd.PTC_CNTRCT_NOV_ID,
-                rcd.PTC_PBP_NOV_ID,
-                rcd.PTC_PLAN_TYPE_NOV_CD,
-                rcd.PTD_CNTRCT_NOV_ID,
-                rcd.PTD_PBP_NOV_ID,
-                rcd.PTD_SGMT_NOV_ID,
-                rcd.RDS_NOV_IND,
-                rcd.CST_SHR_GRP_NOV_CD,
-                rcd.META_DUAL_ELGBL_STUS_NOV_CD);
-                
-            -- Dec
-            call update_bene_monthly(
-                rcd.BENE_ID,
-                TO_DATE(rcd.RFRNC_YR || '-12-01', 'YYYY-MM-DD'),
-                rcd.FIPS_STATE_CNTY_DEC_CD,
-                rcd.MDCR_STUS_DEC_CD,
-                rcd.MDCR_ENTLMT_BUYIN_12_IND,
-                rcd.HMO_12_IND,
-                rcd.PTC_CNTRCT_DEC_ID,
-                rcd.PTC_PBP_DEC_ID,
-                rcd.PTC_PLAN_TYPE_DEC_CD,
-                rcd.PTD_CNTRCT_DEC_ID,
-                rcd.PTD_PBP_DEC_ID,
-                rcd.PTD_SGMT_DEC_ID,
-                rcd.RDS_DEC_IND,
-                rcd.CST_SHR_GRP_DEC_CD,
-                rcd.META_DUAL_ELGBL_STUS_DEC_CD);
-                
-            rcd_cnt := rcd_cnt + 1;
+                IF NOT FOUND
+                THEN
+                    RAISE INFO 'No more records to process....';
+                    okToCont := false;
+                    EXIT inner;
+                END IF;
+
+                -- try for some efficiency w/ monthly dates since the cursor
+                -- will have exec'd a simple fetch (no index used) and we know
+                -- the structure of the table is all 2019 rcds then 2020 rcds;
+                -- so (if lucky) we may just calc dates twice!
+                IF rcd.RFRNC_YR <> currYear
+                THEN
+                    RAISE INFO 'Updating monthly date cache for: %', rcd.RFRNC_YR;
+                    Jan1 := TO_DATE(rcd.RFRNC_YR || '-01-01', 'YYYY-MM-DD');
+                    Feb1 := TO_DATE(rcd.RFRNC_YR || '-02-01', 'YYYY-MM-DD');
+                    Mar1 := TO_DATE(rcd.RFRNC_YR || '-03-01', 'YYYY-MM-DD');
+                    Apr1 := TO_DATE(rcd.RFRNC_YR || '-04-01', 'YYYY-MM-DD');
+                    May1 := TO_DATE(rcd.RFRNC_YR || '-05-01', 'YYYY-MM-DD');
+                    Jun1 := TO_DATE(rcd.RFRNC_YR || '-06-01', 'YYYY-MM-DD');
+                    Jul1 := TO_DATE(rcd.RFRNC_YR || '-07-01', 'YYYY-MM-DD');
+                    Aug1 := TO_DATE(rcd.RFRNC_YR || '-08-01', 'YYYY-MM-DD');
+                    Sep1 := TO_DATE(rcd.RFRNC_YR || '-09-01', 'YYYY-MM-DD');
+                    Oct1 := TO_DATE(rcd.RFRNC_YR || '-10-01', 'YYYY-MM-DD');
+                    Nov1 := TO_DATE(rcd.RFRNC_YR || '-11-01', 'YYYY-MM-DD');
+                    Dec1 := TO_DATE(rcd.RFRNC_YR || '-12-01', 'YYYY-MM-DD');
```

Comment:
This is a nice change readability change in addition to the potential perf improvement from only solving it once.

Reply:
Well the old way was 130+ M * 12 == A VERY BIG NUMBER!

So I was looking for any/all improvements 'cuz this thing could run too long anyways!